homework/index.js

```diff
@@ -28,26 +30,133 @@
+    createAndAppend('th', tr, { text: body });
+    createAndAppend('td', tr, { text: value });
+    return tr;
+  }
+
+  function renderRepoDetails(repo) {
+    const leftSide = document.getElementById('left-side');
+    leftSide.innerHTML = '';
+    const ul = createAndAppend('ul', leftSide);
+    const li = createAndAppend('li', ul);
+    const table = createAndAppend('table', li);
+    const tr1 = addTable(table, 'Repository:', '');
+    createAndAppend('a', tr1.lastChild, {
+      href: repo.html_url,
+      text: repo.name,
+    });
+    addTable(table, 'Description:', repo.description);
+    addTable(table, 'Fork: ', repo.forks);
+    addTable(table, 'Updated:', formatDate(repo.updated_at));
+  }
+
+  function fetchAndRender(repo) {
+    const rightSide = document.getElementById('right-side');
+    rightSide.innerHTML = '';
+    fetchJSON(repo.contributors_url)
+      .then(contributors => {
+        createAndAppend('span', rightSide, {
+          text: 'Contributions',
+          class: 'title',
         });
-        return;
-      }
-      const ul = createAndAppend('ul', root);
-      repos.forEach(repo => renderRepoDetails(repo, ul));
+        const contributorList = createAndAppend('ul', rightSide, {
+          class: 'contributor-list',
+        });
+        contributors.forEach(contributor => {
+          const list = createAndAppend('li', contributorList, {
+            class: 'contributor-item',
+          });
+          createAndAppend('img', list, {
+            class: 'image',
+            src: contributor.avatar_url,
+          });
+          createAndAppend('span', list, {
+            class: 'contributor-name',
+            text: contributor.login,
+          });
+          createAndAppend('span', list, {
+            class: 'contribution-count',
+            text: contributor.contributions,
+          });
+          list.addEventListener('click', () => {
+            window.open(contributor.html_url, '_blank');
+          });
+          list.addEventListener('key', event => {
+            if (event.key === 'Enter')
+              window.open(contributor.html_url, '_blank');
+          });
+        });
+      })
+      .catch(error => renderError(error));
+  }
+
+  function startPage(repository) {
+    const select = document.getElementById('select');
+    repository.sort((a, b) => a.name.localeCompare(b.name));
+    repository.forEach((repositories, index) => {
+      createAndAppend('option', select, {
+        value: index,
+        text: repositories.name,
+      });
+    });
+    renderRepoDetails(repository[select.value]);
+
+    select.addEventListener('change', () => {
+      renderRepoDetails(repository[select.value]);
+      fetchAndRender(repository[select.value]);
     });
+    return repository[select.value];
   }
 
+  function main(url) {
+    const root = document.getElementById('root');
+    const header = createAndAppend('header', root);
+    createAndAppend('p', header, {
+      class: 'Header',
```

Comment:
CSS names should be in **kebab-case**: all lowercase (and with hyphens to separate words).